harvest-finance/src/mapping.ts

```diff
@@ -0,0 +1,213 @@
+  if(addressERC20InputResponse.reverted) {
+    return;
+  } 
+  let addressERC20Input = addressERC20InputResponse.value;
+
+
+  let pricePerShareResponse = contract.try_getPricePerFullShare();
+  if(pricePerShareResponse.reverted) {
+    return;
+  }
+  let pricePerShare = pricePerShareResponse.value;
+
+  let underlyingUnitResponse = contract.try_underlyingUnit();
+  if(underlyingUnitResponse.reverted) {
+    return;
+  }
+  let underlyingUnit = underlyingUnitResponse.value;
+
+  let outputTokenBalanceResponse = contract.try_balanceOf(event.params.to);
+  if(outputTokenBalanceResponse.reverted) {
+    return;
+  }
+  let outputTokenBalance = outputTokenBalanceResponse.value;
+
+  let inputTokenAmount = event.params.value.times(pricePerShare).div(underlyingUnit);
+  let inputTokenBalance = outputTokenBalance.div(pricePerShare).times(underlyingUnit)
+
+  let inputTokenAmounts: TokenBalance[] = [
+    new TokenBalance(addressERC20Input.toHexString(), event.params.to.toHexString(), inputTokenAmount)
+  ];
+  
+  // nothing claimed on deposit
+  let rewardTokenAmounts: TokenBalance[] = [];
+  let rewardTokenBalances: TokenBalance[] = [];
+
+
+  let inputTokenBalances: TokenBalance[] = [
+    new TokenBalance(addressERC20Input.toHexString(), event.params.to.toHexString(), inputTokenBalance)
+  ];
+
+  let outputTokenAmount = event.params.value;
+  if(event.params.from.toHexString() == ADDRESS_ZERO) {
+    // mint fAsset / deposit / investInMarket
+    let receiver = getOrCreateAccount(event.params.to);    
+    investInMarket(
+      event,
+      receiver,
+      market,
+      outputTokenAmount,
+      inputTokenAmounts,
+      rewardTokenAmounts,
+      outputTokenBalance,
+      inputTokenBalances,
+      rewardTokenBalances,
+      null
+    );
+
+  } else if(event.params.to.toHexString() == ADDRESS_ZERO) {
+    // burn fAsset / withdraw / redeemFromMarket
+    let sender = getOrCreateAccount(event.params.from);
+    redeemFromMarket(
+      event,
+      sender,
+      market,
+      outputTokenAmount,
+      inputTokenAmounts,
+      rewardTokenAmounts,
+      outputTokenBalance,
+      inputTokenBalances,
+      rewardTokenBalances,
+      null
+    )
+  } else {
+    // simple transfer
+    let sender = getOrCreateAccount(event.params.from);
+    redeemFromMarket(
+      event,
+      sender,
+      market,
+      outputTokenAmount,
+      inputTokenAmounts,
+      rewardTokenAmounts,
+      outputTokenBalance,
+      inputTokenBalances,
+      rewardTokenBalances,
+      null
+    );
+
+    let receiver = getOrCreateAccount(event.params.to);    
+    investInMarket(
+      event,
+      receiver,
+      market,
+      outputTokenAmount,
+      inputTokenAmounts,
+      rewardTokenAmounts,
+      outputTokenBalance,
+      inputTokenBalances,
+      rewardTokenBalances,
+      null
+    );
+  }
+}
+
+export function handleDoHardWork(call: DoHardWorkCall): void {
+  // create fake event
```

Comment:
You can simplify it this way

  let event = new ethereum.Event();
  event.block = call.block;
  event.transaction = call.transaction;

Reply:
not possible with apiVersion 0.0.6 :(